modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1972,6 +1976,77 @@ public void testGetObjectEntryRelatedObjectEntriesWithAccountEntryRestricted()
 			objectRelationship2.getObjectRelationshipId());
 	}
 
+	@Test
+	public void testPartialUpdateObjectEntry() throws Exception {
+		ObjectEntry objectEntry = _defaultObjectEntryManager.addObjectEntry(
+			_dtoConverterContext, _objectDefinition2,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"dateObjectFieldName", "2023-08-10"
+					).put(
+						"decimalObjectFieldName", 2.7
+					).put(
+						"integerObjectFieldName", 50
+					).put(
+						"longIntegerObjectFieldName", 30000L
+					).put(
+						"precisionDecimalObjectFieldName",
+						new BigDecimal(
+							0.9876543217654321, MathContext.DECIMAL64)
+					).put(
+						"richTextObjectFieldName",
+						StringBundler.concat(
+							"<i>", RandomTestUtil.randomString(), "</i>")
+					).put(
+						"textObjectFieldName", RandomTestUtil.randomString()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		_partialUpdateObjectEntryValues(
+			_objectDefinition2, objectEntry.getId(),
+			HashMapBuilder.<String, Object>put(
+				"integerObjectFieldName", 25
+			).put(
+				"longIntegerObjectFieldName", 200L
+			).put(
+				"precisionDecimalObjectFieldName",
+				new BigDecimal(0.8755445767, MathContext.DECIMAL64)
+			).put(
+				"richTextObjectFieldName", "<i>richText</i>"
+			).put(
+				"textObjectFieldName", "john"
+			).build());
+
+		Calendar calendar = new GregorianCalendar();
+
+		calendar.set(Calendar.YEAR, 2023);
+		calendar.set(Calendar.MONTH, Calendar.AUGUST);
+		calendar.set(Calendar.DAY_OF_MONTH, 10);
+		calendar.set(Calendar.HOUR_OF_DAY, 0);
+		calendar.set(Calendar.MINUTE, 0);
+		calendar.set(Calendar.SECOND, 0);
+		calendar.set(Calendar.MILLISECOND, 0);
+
+		Timestamp timestamp = new Timestamp(calendar.getTimeInMillis());
```

Comment:
@pedro-oliveira446 you can try to use the `SimpleDateFormat.parse` to simplify this.